README.md

```diff
@@ -2,4 +2,329 @@
+
+We'll discuss generics in further detail in upcoming lessons. The effect of this is that the compiler gives individual instances of these collections knowledge of their membership types when they are declared, but otherwise the collection types themselves work exactly the same way whether they are told that they hold `Int`s, or `String`s, or anything else.
+
+### `let` versus `var`
+
+Collections declared using `let` are *immutable* and cannot be altered at all after they are initialized. If you wish to make a collection *mutable*, declare it with `var`. Apple recommends declaring collections with `let` as a default until you encounter the need to alter its contents, claiming that `let` allows the compiler to streamline performance if the collection never needs to be altered.
+
+### Pass-by-value
+
+Swift's collections types are passed by *value* and not by *reference*. When a collection instance is assigned to another collection instance, a copy of the instance is generated and assigned to the new instance. The effect of this is important to understand: **changes made to a collection after it is passed are not reflected in the state of the original collection.**
+
+This is true both for local instances and instances passed into functions and methods via arguments. *If you wish to preserve the mutated state of an instance that is passed-by-value, you have to assign the mutated version back to the original instance.*
+
+## `Array`
+
+### Type Annotation
+
+To explicitly declare the type of an empty array, wrap the type within square brackets `[` `]`. You can use either type annotation to set it equal to empty square brackets, or the initializer syntax that uses a parenthesis following the array type:
+
+```swift
+let array: [Element] = []      // type annotation
+let array: array<Element> = [] // long form annotation
+let array = [Element]()        // initializer
+```
+
+To explicitly declare an array with contents, the contents must match the annotated type.
+
+```swift
+var instructors: [String] = ["Joe", "Tim", "Jim", "Tom"]
+let primes: [Int] = [1, 2, 3, 5, 7, 11]
+let sqrts: [Double] = [1, 1.414, 1.732, 2.236, 2.646, 3.317]
+```
+
+### Subscripting
+
+Arrays can be subscripted by passing an index integer into a pair of square brackets suffixed to the array's name:
+
+```swift
+// getting
+
+let joe = instructors[0]
+print(joe)
+print(instructors[1])
+```
+This will print:
+
+```
+Joe
+Tim
+```
+
+```swift
+// setting, only for arrays declared with var
+
+instructors[2] = "Jim Campagno"
+primes[5] = 13   // error
+```
+### Methods
+
+#### `.append()`
+
+You can add to mutable array using the `.append()` method with the element to add placed inside the parenthesis. The element must match the array's type.
+
+```swift
+instructors.append("Mark")
+instructors.append(42)   // error
```

Comment:
Can you show the error text instead of just saying error? It maaaay change in future versions of swift but hopefully by Swift 2 it won't change too much